buildSrc/src/main/java/GitStatsTask.kt

```diff
@@ -0,0 +1,77 @@
+import com.esotericsoftware.yamlbeans.YamlConfig
+import com.esotericsoftware.yamlbeans.YamlWriter
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import java.io.FileInputStream
+import java.io.FileWriter
+import java.io.StringWriter
+import java.nio.charset.StandardCharsets
+
+/** Counts the occurance of changes in git commits for a given regex and
+ *  writes the result in a YML file. */
+open class GitStatsTask : DefaultTask() {
+    @get:Input lateinit var targetFile: String
+    @get:Input lateinit var commitFileFilter: Regex
+    @get:Input var binaryScore: Int = 0
+    @get:Input var commitSkipList: Array<String> = arrayOf()
+  
+    @TaskAction fun run() {
+        val countsByName = mutableMapOf<String, Int>()
+        val countsByCommit = mutableMapOf<String, Int>()
+        Runtime.getRuntime().exec("git log --no-merges --pretty='%an'%n%H --numstat").inputStream.bufferedReader().useLines { lines ->
+            var name = ""
+            var commit = ""
+            var commitNext = false
+            var skipNext = false
+            for(line in lines) {
+                if (line.startsWith('\'')) {
+                    name = line.trim('\'')
+                    skipNext = false
+                    commitNext = true
+                } else if (commitNext) {
+                    commit = line
+                    if (line.trim() in commitSkipList) {
+                      println("Found and skipping commit " + line)
+                      skipNext = true
+                    }
+                    commitNext = false
+                } else {
+                    val splits = line.split(Regex("\\s+"))
+                    if (!splits.last().matches(commitFileFilter)) continue
+                    val additions = splits[0].toIntOrNull() ?: binaryScore
+                    val deletions = splits[1].toIntOrNull() ?: binaryScore
+                    val commitTotal = additions + deletions
+                    if (!skipNext) {
+                        countsByName[name] = commitTotal + countsByName.getOrPut(name, { 0 })
+                    }
+                    if (commitTotal > 400) {
+                        countsByCommit[commit] = commitTotal
+                    }
```

Comment:
What is this for?

Reply:
Just for stats, to list any commits with lots of changes to try and find more of those suspect ones. Could be removed in due course once we're happy nothing suspect is getting through.